src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Validator.cs

```diff
@@ -145,15 +149,31 @@ private interface IBase64Validatable<T>
             public static int IndexOfAnyExcept(ReadOnlySpan<char> span) => span.IndexOfAnyExcept(s_validBase64Chars);
             public static bool IsWhiteSpace(char value) => Base64.IsWhiteSpace(value);
             public static bool IsEncodingPad(char value) => value == EncodingPad;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static bool ValidateAndDecodeLength(int length, int paddingCount, out int decodedLength) =>
+                Base64ByteValidatable.ValidateAndDecodeLength(length, paddingCount, out decodedLength);
         }
 
         private readonly struct Base64ByteValidatable : IBase64Validatable<byte>
         {
-            private static readonly SearchValues<byte> s_validBase64Chars = SearchValues.Create(EncodingMap);
+            private static readonly SearchValues<byte> s_validBase64Chars = SearchValues.Create(Base64EncoderByte.EncodingMap);
 
             public static int IndexOfAnyExcept(ReadOnlySpan<byte> span) => span.IndexOfAnyExcept(s_validBase64Chars);
             public static bool IsWhiteSpace(byte value) => Base64.IsWhiteSpace(value);
             public static bool IsEncodingPad(byte value) => value == EncodingPad;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static bool ValidateAndDecodeLength(int length, int paddingCount, out int decodedLength)
+            {
+                if (length % 4 == 0)
```

Comment:
Just to point out that if you change other places to `Math.DivRem` keep this check here as is, because in machine code for this a fast `test reg, 3` is created.